aptos-move/aptos-memory-usage-tracker/src/lib.rs

```diff
@@ -67,10 +64,107 @@ where
 
     #[inline]
     fn release_heap_memory(&mut self, amount: AbstractValueSize) {
-        if self.feature_version() >= 3 {
+        if self.feature_version >= 3 {
             self.memory_quota += amount;
         }
     }
+
+    #[inline]
+    fn current_memory_usage(&self) -> AbstractValueSize {
+        match self.initial_memory_quota.checked_sub(self.memory_quota) {
+            Some(usage) => usage,
+            None => AbstractValueSize::zero(),
+            // Note: It's possible for the available memory quota to rise above the initial quota
+            //       under rare circumstances (e.g. values not being tracked initially).
+            //       In such cases, just treat the usage as 0.
+        }
+    }
+}
+
+pub struct ExtendedMemoryAlgebra {
```

Comment:
I think we should find a better name for this :)

Reply:
Yeah, open to suggestions on this one!